hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -460,24 +460,24 @@ protected void postCommit(HoodieTable<T, I, K, O> table, HoodieCommitMetadata me
   }
 
   protected void runTableServicesInline(HoodieTable<T, I, K, O> table, HoodieCommitMetadata metadata, Option<Map<String, String>> extraMetadata) {
-    if (config.inlineTableServices()) {
+    if (config.inlineTableServices() || config.scheduleAsyncTableServices()) {
       if (config.isMetadataTableEnabled()) {
         table.getHoodieView().sync();
       }
       // Do an inline compaction if enabled
-      if (config.inlineCompactionEnabled()) {
+      if (config.inlineCompactionEnabled() || config.scheduleAsyncCompaction()) {
         runAnyPendingCompactions(table);
         metadata.addMetadata(HoodieCompactionConfig.INLINE_COMPACT.key(), "true");
```

Comment:
Why we still need to run the pending compaction and clustering plans ?

Reply:
I had some local changes which I missed to update here. essentially, we can schedule only if there are no other pending jobs. Will push my changes along with addressing other feedback as well.